op-node/sources/caching/cache.go

```diff
@@ -0,0 +1,43 @@
+package caching
+
+import lru "github.com/hashicorp/golang-lru"
+
+type Metrics interface {
+	CacheAdd(label string, cacheSize int, evicted bool)
+	CacheGet(label string, hit bool)
+}
+
+// LRUCache wraps hashicorp *lru.Cache and tracks cache metrics
+type LRUCache struct {
+	m     Metrics
+	label string
+	inner *lru.Cache
+}
+
+func (c *LRUCache) Get(key any) (value any, ok bool) {
```

Comment:
I know the underlying cache is using `any`, but would we able to switch to generics here & enforce some amount of type safety? I we don't mix types in our caches

Reply:
Tried, it requires casting the inner LRU value type back from an interface to the `V` type param, which Go doesn't support. Switching to another LRU lib that supports generics isn't a great option either: the most popular libs I could find on github wrap the LRU contents more naively with sync methods, or don't expose the evictions / cache size. The library we use now is more mature w.r.t. these features. So I'm going to stick with that. We can switch LRU to generics in a different PR.